docs/modules/ROOT/pages/explanation/decisions/usernames.adoc

```diff
@@ -0,0 +1,57 @@
+* Use the `Username` property of Keycloak user objects.
++
+With this approach we can use the existing https://github.com/redhat-cop/group-sync-operator[redhat-cop/group-sync-operator] as it uses the Keycloak `Username` property to populate the user list in the OpenShift `Group` objects it creates.
+Additionally, `oc whoami` provides users with the `Username` they chose when registering their user.
+However, Users can't change their `Username` after registration, as the OpenShift `User` and `Identity` objects, which are created when the users first logs in, aren't updated when the user changes their `Username` in Keycloak.
+
+* Use the `ID` property of Keycloak user objects.
++
+With this approach, users can freely change their `Username` in Keycloak.
++
+However, their `Username` doesn't appear anywhere in the OpenShift cluster with this approach and the output of `oc whoami` is the Keycloak user's `ID`.
+Additionally, group synchronization from Keycloak to OpenShift will require a custom implementation to create `Group` objects which are populated with the group members' `ID` properties, as the group-sync operator currently only supports using the Keycloak user's `Username` field when creating `Group` objects.
+
+* Use the `Email` property of Keycloak user objects.
++
+With this approach, users can't change their E-Mail address after signing up, for the same reasons as they can't change their `Username` property when using that property (see first alternative).
+
+
+Decision::
+* We use the `Username` property of Keycloak user objects as the name for the resulting OpenShift `User` objects
+* We allow users to freely choose their `Username` when signing up
+* We disallow users to change their `Username` after creation of the user in Keycloak
+
+Rationale::
+
+Regardless of the choice of identifier to determine how the OpenShift `User` object is named, the user's identity in the OpenShift cluster is always tied to their Keycloak user's `ID` property.
+This is because Openshift's OpenID Connect identity provider uses the OpenID claim `sub` to establish the user's identity, see the https://docs.openshift.com/container-platform/4.8/authentication/identity_providers/configuring-oidc-identity-provider.html[OpenShift documentation].
+The OpenID claim `sub` is mapped to the Keycloak user's `ID` by default.
++
+For all described alternatives, Keycloak can be configured to require that the property must be unique per realm.
+Therefore, we don't risk having OpenShift `User` objects which are associated with multiple Keycloak users with any of the described alternatives.
++
+First, we decide to use the `Username` property as the identifier used for the OpenShift `User` object.
+This choice provides the best trade-off in our eyes, as it allows users to change their E-Mail address at any time and it shows users a human-verifiable output when running `oc whoami`.
++
+Second, we decide to allow users to freely choose their `Username` when signing up.
```

Comment:
This sentence doesn't provide the reason why we chose it like this.
```suggestion
Second, allowing users to freely choose their `Username` when signing up increases user experience since they can choose a memorable name they like.
```